src/execution_tree/primitives/column_slicing.cpp

```diff
@@ -93,6 +93,26 @@ namespace phylanx { namespace execution_tree { namespace primitives
                 // column = col_start
                 // n = (col_stop - col_start)
 
+                if (col_start < 0 && col_stop >= 0)    //slice from the end
```

Comment:
What is the intended semantic for `start < 0 && stop => 0`? I'd have expected for this to be an error.

Reply:
I was thinking for `col_start < 0 && col_stop >=0`  if a user wants say get the last two column, he can simply say column_slice(a, -2, x) , here, x is any value greater than or equal to zero, that is any positive value  can be passed for the argument col_stop.

Reply:
Do you mean that if `start < 0 && end >= 0`, then the `end` specifies the number of columns instead of the index of the first column after the last?

Reply:
if `start < && end >= 0` , whatever end specifies is ignored internally and the value passed to start is used to extract the number of columns. 
For example,
if the user wants to extract the last 3 columns from the input array, he would say 
`column_slice(a, -3, some_positive_value_ignored_internally)`.
But if the user wants to extract two columns before the last one, 
`column_slice(a, -3, -1)`

Reply:
Ok. May I suggest not to implement this API an such an inconsistent way? This seems to be very confusing and does not follow the 'rule of the least surprise'.

Reply:
Yes sir, how about I throw an error if `start <0 and stop >0` ? If the user wants to extract the last two columns he will have to say `column_slice(a, -2, 0)`  and if the user wants to extract two columns before the last one, `column_slice(a, -3,-1)`. Will that be an acceptable solution ?

Reply:
That sounds about right. Thanks!